lib/modules/manager/cargo/artifacts.ts

```diff
@@ -29,6 +29,28 @@ async function cargoUpdate(
   await exec(cmd, execOptions);
 }
 
+async function cargoUpdatePrecise(
+  manifestPath: string,
+  updatedDeps: Upgrade<Record<string, unknown>>[],
+  constraint: string | undefined
+): Promise<void> {
+  const cmds = updatedDeps
+    .filter((dep) => !!dep.lockedVersion)
+    .map(
+      (dep) =>
+        `cargo update --manifest-path ${quote(manifestPath)}` +
+        ` --package ${dep.packageName!}@${dep.lockedVersion}` +
+        ` --precise ${dep.newVersion}`
```

Comment:
@viceice How do you feel about doing something like this to handle dependencies that have already been updated to the new version? The alternative would be to add recursive retries to `updateArtifacts`, but I feel like that's way more complicated.

```suggestion
        // Hack: If a package is already at `newVersion` then skip updating.
        `cargo update --manifest-path ${quote(manifestPath)}` +
        ` --package ${dep.packageName!}@${dep.newVersion}` +
        ` --precise ${dep.newVersion}` +
        ' > /dev/null 2>&1 ||' +
        // Otherwise update the package to `newVersion`.
        ` cargo update --manifest-path ${quote(manifestPath)}` +
        ` --package ${dep.packageName!}@${dep.lockedVersion}` +
        ` --precise ${dep.newVersion}`
```

Reply:
Got no response so I went ahead and implemented it like this.